creator-node/sequelize/migrations/20200623002237-add-parent-file-data.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+/**
+ * Adds two columns to the `files` table
+ * - fileName: the actual queryable source filename stripped of any prefixes
+ * - dirMultihash: if the file is in an IPFS directory, the CID/multihash for the parent dir
+ */
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    const transaction = await queryInterface.sequelize.transaction()
+    try {
+      await queryInterface.addColumn(
+        'Files',
+        'fileName',
+        {
+          type: Sequelize.TEXT,
+          allowNull: true
+        },
+        { transaction }
+      )
+      await queryInterface.addColumn(
+        'Files',
+        'dirMultihash',
+        {
+          type: Sequelize.TEXT,
+          allowNull: true
+        },
+        { transaction }
+      )
+
+      const files = (await queryInterface.sequelize.query(`SELECT * FROM "Files";`, { transaction }))[0]
+      for (let file of files) {
+        // Set the filename to the last uri "part" of the source file if present
+        // otherwise, use just the sourceFile
+        const fileName = file.sourceFile
+          ? file.sourceFile.split('/').slice(-1)[0] || file.sourceFile
+          : file.sourceFile
+
+        // Set the dir multihash to the parent CID if there is one
+        let dirMultihash
+        if (file.type === 'image' && file.storagePath) {
+          const storagePathParts = file.storagePath.split('/')
+          if (storagePathParts.length === 7) {
```

Comment:
when is the length ever = 7? do you mean 4 here?
i dont think this works?

Reply:
FWIW -- looks like it's different locally vs. in the containers on staging/prod, which is why i had 7. needs another look obv, but separating the migration. though i think we should do it soon